web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,341 @@
+  private static readonly ABANDONED_STATE_TAG: number = 19;
+  private static readonly STATE_TO_COLOR: { [index: number]: string } = {
+    5: '#34a853',
+    6: '#d50000',
+    10: '#2196f3',
+    12: '#d50000',
+    15: '#2196f3',
+    16: '#d50000'
+  };
+  
+  ngOnInit(): void { }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!changes.pushInfos.isFirstChange() && !this.svg) {
+      this.populateData(changes.pushInfos.currentValue);
+      this.initialChart();
+      this.updateChart(0); // Initialize the focus chart with dataAll.
+      // TODO: Call function that creats the brushSelector.
+    }
+  }
+
+  /**
+   * This function populates pushID, state, startTime and duration of all 
+   * pushes for dataAll and of complete pushes for dataComplete. Both arrays
+   * are in ascending order by startTime.
+   * @param pushInfos Array for one push def
+   */
+  private populateData(pushInfos: step189_2020.IPushInfo[]): void {
+    if (!pushInfos) { return; }
+    pushInfos.reverse().forEach(pushInfo => {
+      if (!pushInfo) { return; }
+      const states = pushInfo.stateInfo;
+      if (!states) { return; }
+      const pushID = pushInfo.pushHandle;
+      if (!pushID) { return; }
+      const endState = states[states.length - 1].state;
+      if (!endState) { return; }
+      const pushStartTime = states[0].startTimeNsec;
+      if (!pushStartTime) { return; }
+      const pushEndTime = states[states.length - 1].startTimeNsec;
+      if (!pushEndTime) { return; }
+
+      // Filter pushes with only one state (0 duration) and endState which 
+      // should not be considered.
+      if (states.length > 1) { 
+        if (endState !== BarChartComponent.ABANDONED_STATE_TAG 
+          && endState !== BarChartComponent.BUILT_STATE_TAG
+          && endState !== BarChartComponent.BUILT_UNCHANGED_TAG
+          && endState !== BarChartComponent.SET_DEFAULT_VERSION_STATE_TAG) {
+          const startTime = formatDate((+pushStartTime / BarChartComponent.NANO_TO_MILLI), 
+            BarChartComponent.DATE_FORMAT, BarChartComponent.DATE_LOCALE);
+          const duration = (+pushEndTime - +pushStartTime) / 
+            BarChartComponent.NANO_TO_SECS / BarChartComponent.SECS_TO_HRS;
+          const thePush: BarChartData = {
+            pushID,
+            state: endState,
+            startTime,
+            duration
+          };
+          this.totalDuration.push(duration);
+          this.dataAll.push(thePush);
+          
+          if (endState === BarChartComponent.COMPLETE_STATE_TAG) {
+            this.dataComplete.push(thePush);
+            this.completeDuration.push(duration);
+          }
+        }
+      }
+    })
+  }
+
+  /**
+   * This function scales the x-axis and y-axis and initializes empty focus 
+   * and brush elements.
+   * 
+   * Structure of the SVG:
+   * focus:
+   *   - yAxisFocus, xAxisFocus
+   *   - title of y-axis
+   * brush:
+   *   - xAxisBrush
```

Comment:
Can you indicate the tags and a classes? An expanded version of this:

    <svg>
      <g>
        // Top plot.
        <g>
          [...]
        </g>
        <text>
           // y-axis title.
        </text>
      </g>
      <g>
        // Bottom plot.
      </g>
    <svg>